homeassistant/components/light/yeelight.py

```diff
@@ -286,13 +327,63 @@ def set_flash(self, flash) -> None:
             except BulbException as ex:
                 _LOGGER.error("Unable to set flash: %s", ex)
 
+    @_cmd
+    def set_effect(self, effect) -> None:
+        """Activate effect."""
+        if effect:
+            from yeelight import (Flow, BulbException)
+            from yeelight.transitions import (disco, temp, strobe, pulse,
+                                              strobe_color, alarm, police,
+                                              police2, christmas, rgb,
+                                              randomloop, slowdown)
+            # transition = int(self.config[CONF_TRANSITION])
+            if effect == EFFECT_DISCO:
+                flow = Flow(count=0, transitions=disco(),)
```

Comment:
Probably my incompetence. The reason i have this is because this is how i did it to get it to work. the disco function returns a list with the transitions. I just tried you example [disco()] but i can't get it to work. maybe i need to do something other then replace disco() with [disco()] ?